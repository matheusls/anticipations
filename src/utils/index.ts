export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
