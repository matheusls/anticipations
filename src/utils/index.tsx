import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
