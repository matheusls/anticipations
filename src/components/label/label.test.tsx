import { render, screen } from 'tests/utils';

import Label from './label';

describe('<Label />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Label htmlFor="some-input">Label component</Label>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Label correctly', () => {
    render(<Label htmlFor="some-input">Label component</Label>);

    expect(screen.getByText(/label component/i)).toBeInTheDocument();
  });
});
