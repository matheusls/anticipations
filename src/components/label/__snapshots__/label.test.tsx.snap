// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should match snapshot 1`] = `
.c0 {
  color: #656565;
  display: block;
  font-size: 0.875rem;
  font-weight: 400;
  margin-bottom: 6px;
}

<label
  class="c0"
  for="some-input"
>
  Label component
</label>
`;
